master/docs/manual/customization.rst

```diff
@@ -567,6 +571,21 @@ See :class:`buildbot.worker.ec2.EC2LatentWorker` for an example.
         If ``fast`` is ``True`` then the function should call back as soon as it is safe to do so, as, for example, the master may be shutting down.
         The value returned by the callback is ignored.
 
+    .. py:attribute:: builds_may_be_incompatible
+
+        Determines if new instances have qualities dependent on the build.
+        If ``True``, the master will call ``is_compatible_with_build`` to determine whether new builds are compatible with the started instance.
+        By default, this is ``False``.
+
+    .. py:method:: is_compatible_with_build(self, build_props)
+
+        This method determines whether a started instance is compatible with the build that is about to be started.
+        ``build_props`` is the properties of the build that are known before the build has been started.
+        If some qualities of the instance are renderable and could depend on the properties of the build, then this method should check whether the current instance has the same qualities as what a new instance started from ``build_props`` would have.
```

Comment:
I think this would be better to just give an example. For me the most obvious example is docker image.